app/models/repository.rb

```diff
@@ -321,27 +323,25 @@ def self.create_from_data(repo_host_data)
     nil
   end
 
-  def check_status
-    Repository.check_status(host_type, full_name)
-  end
-
   # @return [Boolean] false if we received a 404 response code when checking the status, true otherwise
-  def self.check_status(host_type, repo_full_name)
+  def check_status
     domain = RepositoryHost::Base.domain(host_type)
-    response = Typhoeus.head("#{domain}/#{repo_full_name}")
+    response = Typhoeus.head("#{domain}/#{full_name}")
+
+    return true if status == "Hidden" # don't overwrite Hidden projects
 
     if response.response_code == 404
-      repo = Repository.includes(:projects).find_by_full_name(repo_full_name)
```

Comment:
just noticed this bug! We're not limiting the repo to "host_type: host_type", so this might've been updating repos that had the same path but different host types! 🐛